platforms/corda-ent/configuration/setup/vault_kubernetes/tasks/main.yaml

```diff
@@ -0,0 +1,120 @@
+
+###############################################################################################
+# This task enables the path for the nodes on vault
+# This task runs only when {{component_path}} is not already created
+- name: Vault Auth enable for organisation
+  shell: |
+    vault auth enable --path="{{ component_auth }}" kubernetes
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  when: not vault_auth_status   # Only when component_auth is NOT in the output of auth_list
+  ignore_errors: yes
+
+############################################################################################
+# This task get the certificate for the cluster mentioned in k8 secret
+- name: "Get Kubernetes cert files for organizations"
+  shell: |
+    KUBECONFIG={{ kubernetes.config_file }} kubectl -n {{ component_ns }} get secrets $(KUBECONFIG={{ kubernetes.config_file }} kubectl -n {{ component_ns }} get secrets | grep "default" | awk '{print $1}') -o jsonpath="{.data['ca\.crt']}" | base64 -d > "./build/{{ component_ns }}.ca.cert"
+  when: not vault_auth_status
+
+################################################################################################
+# This task writes the Service Account token to the vault for various Corda Ent entity
+- name: Write reviewer token
+  shell: |
+    export REVIEWER_TOKEN=$(KUBECONFIG={{ kubernetes.config_file }} kubectl get secret $(KUBECONFIG={{ kubernetes.config_file }} kubectl get serviceaccount -n {{ component_ns }} vault-reviewer -o jsonpath={.secrets[0].name}) -n {{ component_ns }} -o  jsonpath={.data.token} | base64 -d)
+    vault write auth/{{ component_auth }}/config token_reviewer_jwt="$REVIEWER_TOKEN" kubernetes_host=$(KUBECONFIG={{ kubernetes.config_file }} kubectl config view -o jsonpath="{.clusters[?(@.name==\"{{ kubernetes.context }}\")].cluster.server}") kubernetes_ca_cert=@"./build/{{ component_ns }}.ca.cert"
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  when: not vault_auth_status
+
+# This task checks if the vault-ro polict already exists
+- name: "Checking if the vault policies already exists"
+  include_role:
+    name: "{{ playbook_dir }}/../../shared/configuration/roles/check/setup"
+  vars:
+    check: "vault_policies"
+  register: vault_policy_result
+  ignore_errors: yes
+
+#####################################################################################################################
+# This task creates the access policy for various corda-ent entity
+- name: Create policy for Access Control
+  shell: |
+    cd ./build    
+    echo "path \"secret/{{ component_ns }}/*\" {
+        capabilities = [\"read\", \"list\", \"create\"]
+    }
+    path \"secret/cenm/*\" {
+        capabilities = [\"read\", \"list\", \"create\"]
+    }
+    path \"secret/notary/*\" {
+        capabilities = [\"read\", \"list\", \"create\"]
+    }"  >>  vault-crypto-{{ component_type }}-{{ component_name }}-ro.hcl
```

Comment:
Will it not enable the path for all the org ?

Reply:
Separate policies for each organization

Reply:
Is this conv resolved?

Reply:
Yes, It's already resolved